src/backend/distributed/metadata/metadata_sync.c

```diff
@@ -666,10 +698,97 @@ MetadataCreateCommands(void)
 												  shardCreateCommandList);
 	}
 
+	/* As the last step, propagate the pg_dist_object entities */
+	List *distributedObjectSyncCommandList = DistributedObjectSyncCommandList();
+	metadataSnapshotCommandList = list_concat(metadataSnapshotCommandList,
+											  distributedObjectSyncCommandList);
+
 	return metadataSnapshotCommandList;
 }
 
 
+/*
+ * DistributedObjectSyncCommandList returns the necessary commands to create
+ * pg_dist_object entries on the new node.
+ */
+static List *
+DistributedObjectSyncCommandList(void)
+{
+	HeapTuple pgDistObjectTup = NULL;
+	Relation pgDistObjectRel = table_open(DistObjectRelationId(), AccessShareLock);
+	Relation pgDistObjectIndexRel = index_open(DistObjectPrimaryKeyIndexId(),
+											   AccessShareLock);
+	TupleDesc pgDistObjectDesc = RelationGetDescr(pgDistObjectRel);
+
+	List *objectAddresses = NIL;
+	List *distributionArgumentIndexes = NIL;
+	List *colocationIds = NIL;
+
+	SysScanDesc pgDistObjectScan = systable_beginscan_ordered(pgDistObjectRel,
```

Comment:
Hmm, `systable_beginscan_ordered` is not a familiar API to me. Why do we need to scan ordered? For the tests to be consistent?

Reply:
Yes